### PR TITLE
Added status code to FeignException

### PR DESCRIPTION
**Pull request by FrEaKmAn**
As discussed in #275, I have added status code to FeignException.

### CODE REVIEW

File: core/src/main/java/feign/codec/ErrorDecoder.java [hunk 1]
```diff
@@ -15,6 +15,10 @@
  */
 package feign.codec;
 
+import feign.FeignException;
```
Comment (spencergibb) [inline core/src/main/java/feign/codec/ErrorDecoder.java]: Why move these?

Comment (FrEaKmAn) [inline core/src/main/java/feign/codec/ErrorDecoder.java]: My editor (automatic imports) is doing this. I can revert it.

File: core/src/main/java/feign/FeignException.java [hunk 2]
```diff
@@ -34,6 +35,16 @@ protected FeignException(String message) {
     super(message);
   }
 
+  protected FeignException(int statusCode, String message) {
+    super(message);
+
+    this.statusCode = statusCode;
+  }
+
+  public int getStatusCode() {
```
Comment (adriancole) [inline core/src/main/java/feign/FeignException.java]: Just call this status() (not getStatus/Code) for consistency with other types.

File: core/src/main/java/feign/codec/ErrorDecoder.java (hunk 1 above)
Comment (adriancole) [inline core/src/main/java/feign/codec/ErrorDecoder.java]: Yeah pls do

File: core/src/main/java/feign/FeignException.java [hunk 3]
```diff
@@ -25,6 +25,7 @@
 public class FeignException extends RuntimeException {
 
   private static final long serialVersionUID = 0;
+  private int statusCode;
```
Comment (adriancole) [inline core/src/main/java/feign/FeignException.java]: Final

File: core/src/main/java/feign/FeignException.java [hunk 4]
```diff
@@ -34,6 +35,16 @@ protected FeignException(String message) {
     super(message);
   }
 
+  protected FeignException(int statusCode, String message) {
+    super(message);
+
```
Comment (adriancole) [inline core/src/main/java/feign/FeignException.java]: Nit Remove extra newline

File: core/src/test/java/feign/codec/DefaultErrorDecoderTest.java [hunk 5]
```diff
@@ -71,4 +81,22 @@ public void retryAfterHeaderThrowsRetryableException() throws Throwable {
 
     throw errorDecoder.decode("Service#foo()", response);
   }
+
+  class FeignExceptionStatusCodeMatcher extends TypeSafeMatcher<FeignException> {
```
Comment (adriancole) [inline core/src/test/java/feign/codec/DefaultErrorDecoderTest.java]: Working too hard! Remove this and just do try catch, assert on code. This matcher is distractingly big for the small job it does.

**Comment by adriancole** [comment]
Only style notes. Thanks!

**Comment by adriancole** [comment]
PS when you are finished with style fixups, update CHANGES and make this a single commit (squash)

**Comment by adriancole** [comment]
looks good. once test fail is sorted, can you squash into one commit?

**Comment by adriancole** [comment]
There are 6 commits now :) I'll squash these into one on the way in tomorrow if you're having trouble.

**Comment by FrEaKmAn** [comment]
Yes sorry, it should be fixed now.

**Comment by adriancole** [comment]
great!

**Comment by adriancole** [comment]
ps this will be 8.10 actually

**Pull request merged**